src/test/java/io/anserini/integration/PretokenizedIndexEndToEndTest.java

```diff
@@ -0,0 +1,73 @@
+package io.anserini.integration;
+
+import io.anserini.collection.DocumentCollection;
+import io.anserini.collection.JsonCollection;
+import io.anserini.collection.TrecCollection;
+import io.anserini.index.IndexArgs;
+import io.anserini.index.IndexCollection;
+import io.anserini.index.generator.DefaultLuceneDocumentGenerator;
+import io.anserini.search.SearchArgs;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+
+
+public class PretokenizedIndexEndToEndTest extends EndToEndTest {
+
+    @Override
+    IndexArgs getIndexArgs() {
+        IndexArgs indexArgs = createDefaultIndexArgs();
+
+        indexArgs.input = "src/test/resources/sample_docs/json/collection_tokenized";
+        indexArgs.collectionClass = JsonCollection.class.getSimpleName();
+        indexArgs.generatorClass = DefaultLuceneDocumentGenerator.class.getSimpleName();
+        indexArgs.pretokenized = true;
+        indexArgs.storeRaw = true;
+
+        return indexArgs;
+    }
+
+    @Override
+    protected void setCheckIndexGroundTruth() {
+        docCount = 2;
+        documents.put("2000000", Map.of(
+                "contents", "this was ##a simple pretokenized test",
+                "raw","{\n" +
+                        "  \"id\" : \"2000000\",\n" +
+                        "  \"contents\" : \"this was ##a simple pretokenized test\"\n" +
+                        "}"));
+        documents.put("2000001", Map.of(
+                "contents", "some time extra ##vert ##ing and some time intro ##vert ##ing",
+                "raw","{\n" +
+                        "  \"id\" : \"2000001\",\n" +
+                        "  \"contents\" : \"some time extra ##vert ##ing and some time intro ##vert ##ing\"\n" +
+                        "}"
+        ));
+        tokens.put("2000000",Map.of("contents",
+                Map.of("this",1L, "was",1L,"##a",1L,"simple",1L,"pretokenized",1L,"test",1L)));
+        tokens.put("2000001",Map.of("contents",
+                Map.of("some",2L, "time",2L,"extra",1L,"##vert",2L,"##ing",2L,"and",1L,"intro",1L)));
+
+        fieldNormStatusTotalFields = 1;
+        // whitespace analyzer keeps everything, includes docid
+        // this is ##a simple pretokenized test some time extra ##vert ##ing and intro 2000000 2000001
+        termIndexStatusTermCount = 15;
+        termIndexStatusTotFreq = 15;
+        storedFieldStatusTotalDocCounts = 2;
+        termIndexStatusTotPos = 17 + storedFieldStatusTotalDocCounts;
+        storedFieldStatusTotFields = 6; // 1 docs * (1 id + 1 contents + 1 raw) *2
+    }
+
+    @Override
+    protected void setSearchGroundTruth() {
```

Comment:
do we want to compare every query tokens here?

Reply:
you mean the doc tokens right?

yea, it's not that much to check, e.g.,

> id: 2000000
> "this", "was", "##a", "simple", "pretokenized", "test"

Reply:
oh, no. Sorry for the confusion. I think https://github.com/castorini/anserini/blob/b51f7a14bea31018057aa4907caf7d67657198e0/src/test/java/io/anserini/integration/EndToEndTest.java#L205 has checked every token in the doc along with its freq.

I mean since we also add -pretokenized option to search collection, say we have topics as `"1   ##vert ##ing`, we should expect while we processing this query, we did not stem or filter stopwords. Query tokens while we performing search should be something like `"##vert","##ing"` not `"vert","ing"`. Do we want to check this here? (I think this is the way to test the pretokenized option for search collection?)

Reply:
yes, I think so?